genai-perf/genai_perf/parser.py

```diff
@@ -1168,47 +1164,51 @@ def parse_args():
 
         args = parser.parse_args(config_args)
 
-        config.subcommand = ConfigField(
-            default="profile", value=args.subcommand, required=True
-        )
-
-        # Set verbose
-        config.verbose = ConfigField(default=False, value=args.verbose)
+        config.subcommand = ic.Subcommand(args.subcommand)
+        config.verbose = args.verbose
 
         logger.info(f"Profiling these models: {', '.join(config.model_names)}")
         return args, config, argv[passthrough_index + 1 :]
 
 
 def _create_template_config(args: argparse.Namespace, argv: List[str]) -> ConfigCommand:
-    config = ConfigCommand({"model_name": ""}, skip_inferencing_and_checking=True)
+    config = ConfigCommand(skip_inferencing_and_checking=True)
 
-    config.verbose = ConfigField(
-        default=False, value=args.verbose, add_to_template=False
-    )
-    config.subcommand = ConfigField(
-        default="profile",
-        value=args.subcommand,
-        required=True,
-        add_to_template=False,
-    )
+    config.verbose = args.verbose
+    config.subcommand = ic.Subcommand(args.subcommand)
 
     # The template command is: genai-perf template [filename] [-v/--verbose]
+    del argv[0]  # This is the path to genai-perf
+    del argv[argv.index(ic.Subcommand.TEMPLATE.value)]
     if "-v" in argv:
         del argv[argv.index("-v")]
     if "--verbose" in argv:
         del argv[argv.index("--verbose")]
 
     # Assumption is the final argument is the filename (if it exists)
-    filename = argv[2] if len(argv) > 2 else TemplateDefaults.FILENAME
-    config.template_filename = ConfigField(
-        default=TemplateDefaults.FILENAME, value=filename, add_to_template=False
-    )
+    if argv:
+        config.template_filename = Path(argv[0])
 
     return config
 
 
+def _check_unknown_args(subcommand: str, unknown_args: List[str]) -> None:
+    # Only in the case of the template subcommand do we allow a single unknown arg
+    # which is assumed to be the filename for the templated config being created.
+    if not unknown_args:
+        return
+    elif subcommand != ic.Subcommand.TEMPLATE.value:
```

Comment:
Is this missing a case for pass thru args?

Reply:
No, passthrough args are not included in the list of argv values sent to argparse, so they are never included in the unknown args list:

```
    args, unknown_args = parser.parse_known_args(argv[1:passthrough_index])
```

Reply:
i was conflating the two. thank you